test/functional/managedclusterinfo_test.go

```diff
@@ -214,21 +232,79 @@ var _ = Describe("Metrics", func() {
 					return fmt.Errorf("Unexpect queryResult %v", queryResult)
 				}
 
+				return nil
+			}).Should(BeNil())
+		})
+		By("Query metrics by sum(acm_managed_cluster_labels) by (cloud)", func() {
+			Eventually(func() error {
+				resp, b, err := queryMetrics("sum(acm_managed_cluster_labels)+by+(cloud)")
+				klog.Infof("Get Empty Metrics response: %v", b)
+				if err != nil || resp.StatusCode != http.StatusOK {
+					return fmt.Errorf("Failed to query metrics %v: %v", resp.StatusCode, err)
+				}
+
+				queryResult := QueryResult{}
+				if err := json.Unmarshal([]byte(b), &queryResult); err != nil {
+					return err
+				}
+
+				expectResult := map[string]string{"Amazon": "3", "": "1"}
+				actualResult := map[string]string{}
+				for _, r := range queryResult.Data.Result {
+					key := r.Metric.Cloud + r.Metric.CreatedVia + r.Metric.Vendor
+					actualResult[key] = r.Value[1].(string)
+				}
+
+				klog.Infof("expect result %v", expectResult)
+				klog.Infof("actual result %v", actualResult)
+				if !reflect.DeepEqual(expectResult, actualResult) {
+					return fmt.Errorf("Unexpect queryResult %v", queryResult)
+				}
+
+				return nil
+			}).Should(BeNil())
+		})
+		By("Query metrics by sum(acm_managed_cluster_labels) by (hub_cluster_id)", func() {
+			Eventually(func() error {
+				resp, b, err := queryMetrics("sum(acm_managed_cluster_labels)+by+(hub_cluster_id)")
+				klog.Infof("Get Empty Metrics response: %v", b)
+				if err != nil || resp.StatusCode != http.StatusOK {
+					return fmt.Errorf("Failed to query metrics %v: %v", resp.StatusCode, err)
+				}
+
+				queryResult := QueryResult{}
+				if err := json.Unmarshal([]byte(b), &queryResult); err != nil {
+					return err
+				}
+
+				expectResult := map[string]string{"": "4"}
+				actualResult := map[string]string{}
+				for _, r := range queryResult.Data.Result {
+					key := r.Metric.Cloud + r.Metric.CreatedVia + r.Metric.Vendor
+					actualResult[key] = r.Value[1].(string)
+				}
+
+				klog.Infof("expect result %v", expectResult)
+				klog.Infof("actual result %v", actualResult)
+				if !reflect.DeepEqual(expectResult, actualResult) {
+					return fmt.Errorf("Unexpect queryResult %v", queryResult)
+				}
+
 				return nil
 			}).Should(BeNil())
 		})
 	})
 
 	It("Get Empty Metrics", func() {
 		By("Getting metrics", func() {
-			Eventually(func() string {
+			Eventually(func() int {
```

Comment:
I think we still need to compare the string content, any reason to change to int?